### PR TITLE
Add documentation for API interface

### PR DESCRIPTION
**Pull request by Geod24**
@team : Feel free to make any suggestion / ask any question you see fit.

### CODE REVIEW

**Review by AndrejMitrovic — APPROVED** [review]

**Review by TrustHenry — APPROVED** [review]

File: source/agora/common/API.d [hunk 1]
```diff
@@ -47,7 +47,23 @@ public struct PublicConfig
...
+
+    A full node:
+    - Can connect to any node and request data about the blockchain & network
+    - Accepts external connections and send them blockchain/network data
+    - Store the data it receives on disk
+    - Can catch up with the network when found lagging behind
+    - Validates the data it receives
+    - Receives, stores and forwards transactions (but drop them after a timeout)
+    - Does not participate in consensus
+
```
Comment (TrustHenry) [inline source/agora/common/API.d:62]: - Check the node version.

Comment (TrustHenry) [reply source/agora/common/API.d:62]: - Provide node performance score.

Comment (AndrejMitrovic) [reply source/agora/common/API.d:62]: I think we can add these later? There is a lot of things the node will do, like verifying authenticity of signatures, etc. But we should list these once we add them later. cc @Geod24

Comment (Geod24) [reply source/agora/common/API.d:62]: I tried to focus on blockchain-related stuff. Node performance is not part of those, but the node version might be a good one.

Comment (TrustHenry) [reply source/agora/common/API.d:62]: - HTTP traffic from the outside world

Comment (TrustHenry) [reply source/agora/common/API.d:62]: - Full ledger provided

Comment (TrustHenry) [reply source/agora/common/API.d:62]: - Listening and propagating their decisions blindly to anyone who asks.

**Review by MichaelKim20 — APPROVED** [review]

**Comment by AndrejMitrovic** [comment]
Everyone approved, merging!

**Pull request merged**